src/Kaleidoscope.h

```diff
@@ -147,21 +133,39 @@ class Kaleidoscope_ {
   typedef Key(*eventHandlerHook)(Key mappedKey, byte row, byte col, uint8_t keyState);
   static eventHandlerHook eventHandlers[HOOK_MAX];
 
-  static void replaceEventHandlerHook(eventHandlerHook oldHook, eventHandlerHook newHook);
-  static void appendEventHandlerHook(eventHandlerHook hook);
-  static void useEventHandlerHook(eventHandlerHook hook);
+  static void replaceEventHandlerHook(eventHandlerHook oldHook, eventHandlerHook newHook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.onKeyswitchEvent(...) instead.")));
+  static void appendEventHandlerHook(eventHandlerHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.onKeyswitchEvent(...) instead.")));
+  static void useEventHandlerHook(eventHandlerHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.onKeyswitchEvent(...) instead.")));
 
   typedef void (*loopHook)(bool postClear);
   static loopHook loopHooks[HOOK_MAX];
 
-  static void replaceLoopHook(loopHook oldHook, loopHook newHook);
-  static void appendLoopHook(loopHook hook);
-  static void useLoopHook(loopHook hook);
+  static void replaceLoopHook(loopHook oldHook, loopHook newHook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.beforeEachCycle(), .beforeReportingState(...) or .afterEachCycle(...) instead.")));
+  static void appendLoopHook(loopHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.beforeEachCycle(), .beforeReportingState(...) or .afterEachCycle(...) instead.")));
+  static void useLoopHook(loopHook hook)
+  __attribute__((deprecated("Please implement kaleidoscope::Plugin.beforeEachCycle(), .beforeReportingState(...) or .afterEachCycle(...) instead.")));
 
   static bool focusHook(const char *command);
 };
 
-extern Kaleidoscope_ Kaleidoscope;
+extern kaleidoscope::Kaleidoscope_ Kaleidoscope;
```

Comment:
Class `Kaleidoscope_` is currently ambiguously instanciated both in global namespace and in namespace `kaleidoscope`, [here](https://github.com/keyboardio/Kaleidoscope/blob/af208b0fd35b16531592a08db272acd41a654f7e/src/Kaleidoscope.cpp#L105) and [there](https://github.com/keyboardio/Kaleidoscope/blob/af208b0fd35b16531592a08db272acd41a654f7e/src/Kaleidoscope.cpp#L200).

One of them needs to be removed (preferably the one at global scope).